### ISSUE TITLE
Add "line-aligned" to react/jsx-closing-tag-location

### ISSUE THREAD

**Issue by Cooya** [jsx-eslint/eslint-plugin-react#3740]
It would be quite nice to have the option "line-aligned" in the rule _react/jsx-closing-tag-location_, same as in the rule [https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-closing-bracket-location.md](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-closing-bracket-location.md).

With this option, the following code would be accepted :
```jsx
const test = <div>
	Hello World
</div>;
```
And this one would be denied :
```jsx
const test = <div>
	Hello World</div>;
```

**Comment by ljharb** [comment]
I'm confused - how does closing-bracket-notation not already cover this case?

**Comment by Cooya** [comment]
With `'react/jsx-closing-bracket-location': ['error', 'tag-aligned']` in eslintrc.js, this is accepted :
```jsx
const test = <div>
	Hello World</div>;
```
This as well :
```jsx
<h4><span>
</span></h4>
```
It could be fixed with something like `'react/jsx-closing-tag-location': ['error', 'tag-aligned']`. _react/jsx-closing-tag-location_ without option is too harsh to be used.

**Comment by Cooya** [comment]
Yes, the rule `react/jsx-closing-tag-location` imposes the following syntax :
```jsx
const test = (
	<div>
		Hello World
	</div>
);
```
but I would rather have the possibility to do it without parens :
```jsx
const test = <div>
	Hello World
</div>;
```
It is simpler and shorter.

**Comment by Cooya** [comment]
Yet it is implemented in the very similar rule `jsx-closing-bracket-location` :
```jsx
// 'jsx-closing-bracket-location': [1, 'line-aligned']
var x = <Hello
  firstName="John"
  lastName="Smith"
/>;

var x = function() {
  return <Say
    firstName="John"
    lastName="Smith"
  >
    Hello
  </Say>;
};
```
For me, it makes sense to have the same options in `jsx-closing-tag-location`.

**Comment by ljharb** [comment]
Indeed; it was implemented there before I maintained this plugin.

To be clear, I added the "help wanted" label which means it'll be included once a PR is ready.

**Comment by kimtaejin3** [comment]
@Cooya @ljharb I would love to work on this

**Comment by ljharb** [comment]
@kimtaejin3 go for it
